src/transforms/filter.js

```diff
@@ -8,211 +8,264 @@
 
 'use strict';
 
-var isNumeric = require('fast-isnumeric');
-
-// var Lib = require('@src/lib');
 var Lib = require('../lib');
+var axisIds = require('../plots/cartesian/axis_ids');
 
-/* eslint no-unused-vars: 0*/
+var INEQUALITY_OPS = ['=', '<', '>=', '>', '<='];
+var INTERVAL_OPS = ['[]', '()', '[)', '(]', '][', ')(', '](', ')['];
+var SET_OPS = ['{}', '}{'];
 
-// so that Plotly.register knows what to do with it
 exports.moduleType = 'transform';
 
-// determines to link between transform type and transform module
 exports.name = 'filter';
 
-// ... as trace attributes
 exports.attributes = {
+    enabled: {
+        valType: 'boolean',
+        dflt: true,
+        description: [
+            'Determines whether this filter transform is enabled or disabled.'
+        ].join(' ')
+    },
+    filtersrc: {
+        valType: 'enumerated',
+        values: ['x', 'y', 'z', 'ids'],
```

Comment:
> are we keeping this explicit list of attributes, or is it obsoleted by crawling the attribute tree for filterable arrays?

👍 would be awesome to filter on marker.size, or text, anything that has an array...